pkg/api/v1/resource/helpers.go

```diff
@@ -445,3 +447,34 @@ func MergeContainerResourceLimits(container *v1.Container,
 func IsHugePageResourceName(name v1.ResourceName) bool {
 	return strings.HasPrefix(string(name), v1.ResourceHugePagesPrefix)
 }
+
+// ResourcesEqual returns true if the two resources lists provided are equal.
+func ResourcesEqual(lhs v1.ResourceList, rhs v1.ResourceList) bool {
+	if len(lhs) != len(rhs) {
+		return false
+	}
+	for k, vlhs := range lhs {
+		vrhs, ok := rhs[k]
+		if !ok {
+			return false
+		}
+		if !vlhs.Equal(vrhs) {
+			return false
+		}
+	}
+	return true
+}
+
+// UpdateTotalRequestedResources updates the TotalRequestedResources field.
+// Returns true if pod TotalRequestedResources has changed.
+func UpdateTotalRequestedResources(status *v1.PodStatus, pod *v1.Pod) bool {
+	requests := PodRequests(pod, PodResourcesOptions{
+		InPlacePodVerticalScalingEnabled: utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling),
+	})
+
+	if ResourcesEqual(requests, status.TotalRequestedResources) {
+		return false
```

Comment:
this is the most common code path, correct? Most Pods will only need to have this sent once and will never change. I wonder if there is anything we can do to optimize on this and only update when in-place update happened.

Overhead doesn't seem to be that much though, so should be fine as-is I think